pkgs/development/compilers/gcc/common/libgcc-buildstuff.nix

```
@@ -42,11 +42,14 @@ in
   #
   # No rule to make target '../../../gcc-xx.x.x/libgcc/config/loongarch/crti.S', needed by 'crti.o'.  Stop.
   #
-  # For LoongArch64, a hacky workaround is to simply touch them,
+  # For LoongArch64 and S390, a hacky workaround is to simply touch them,
   # as the platform forces .init_array support.
   #
   # https://www.openwall.com/lists/musl/2022/11/09/3
   #
-  + lib.optionalString stdenv.targetPlatform.isLoongArch64 ''
++ lib.optionalString stdenv.targetPlatform.isLoongArch64 ''
   touch libgcc/config/loongarch/crt{i,n}.S
 ''
++ lib.optionalString stdenv.targetPlatform.isS390 ''
+  touch libgcc/config/s390/crt{i,n}.S
```

```suggestion
+ lib.optionalString (stdenv.targetPlatform.isLoongArch64 || stdenv.targetPlatform.isS390) ''
  touch libgcc/config/${stdenv.targetPlatform.parsed.cpu.family}/crt{i,n}.S
''
```

for these 3 systems the  `parsed.cpu.family` is the same as the name in `config/`
```
nix-repl> :p pkgsCross.loongarch64-linux.stdenv.targetPlatform.parsed.cpu.family
"loongarch"

nix-repl> :p pkgsCross.s390.stdenv.targetPlatform.parsed.cpu.family
"s390"

nix-repl> :p pkgsCross.s390x.stdenv.targetPlatform.parsed.cpu.family
"s390"
```
for riscv it's the same too
but not for aarch64-multiplatform

so it can only be relied on in some cases so we could use it here but with a comment that it's not always correct